### DISCUSSION TITLE
Call multiple time LoadItemsByCamlQueryAsync return wrong result

### DISCUSSION THREAD

**Discussion by necik11** [pnp/pnpcore#946] [General] [answered]
PnP.Core 1.7.0
--

Hello 

I'm not sure if this is the problem or if I'm just using this method incorrectly. But i need call method **LoadItemsByCamlQueryAsync**  several times and i in second time i got result what i have got in 1st call. 
**Description**: I have list - lets call **listA**:

1.  I need find folder on those listA. 
2. Find folders in folder that I found in the previous query.

So what i am doing its:

```
// Get List
var listA= context.Web.Lists.GetByTitle("listA",a => a.RootFolder);
context.Execute();

// Find folder 
var folder = FindFolder(listA);
     public static async Task<IEnumerable<IListItem>> FindFolder(IList list, string folderName)
        {
            string viewXml = $@"
<View >
	<Query>
		<Where>
			<Eq>
				<FieldRef Name='Title'/>
				<Value Type='text'>
					<![CDATA[{folderName}]]>
				</Value>
			</Eq>
		</Where>
	</Query>
</View>
";
            await list.LoadItemsByCamlQueryAsync(new CamlQueryOptions()
            {
                ViewXml = viewXml,
                DatesInUtc = true
            });
            return list.Items.AsRequested();
        }
// Find folders with similiar query, but just add this folder 
3.  var folders = FindFoldersInFolder(listA, folder);
 FolderServerRelativeUrl = folder.ServerRelativeUrl
```

**Expectation:**
The result will be according to the new spQuery/CamlQuery.., 
**Result**
Previous requested items... its work weird.. seems when query not found any result/items its return previous instead of null or 0... 
I will look into this further, any tips would be appreciated

**Comment by jansenbe** [comment, marked as answer]
@necik11 : this is expected behavior. When you use a "Load" method we'll load the data into the in memory `PnPContext` object, for example if you need paging then each page is requested and added so that at the end you've a complete set of rows. If you don't want that behavior the simplest solution is to clear the loaded items before the next invocation of `LoadItemsByCamlQuery` as is shown in here: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-paging-with-loaditemsbycamlquery

![image](https://user-images.githubusercontent.com/7451219/185890256-32fdd21f-ef76-43b2-a4c4-9b42c206e4e7.png)

**Reply by necik11** [reply]
It works!
I would certainly not say that this is expected behavior compared to any version SHP 2010-2019 or CSOM, the items are reloaded when calling LoadItemsByCamlQuery, even you load 0 items... Anyway, clear the loading items works, thanks for fast the answer.
